src/containers/ConversionForm/index.tsx

```diff
@@ -55,15 +59,46 @@ const ConversionForm = ({
     const [fileToConvert, setFileToConvert] = useState<UploadFile>();
     const [engineSelected, setEngineSelected] = useState<boolean>(false);
     const [isProcessing, setIsProcessing] = useState<boolean>(false);
+    const [fileTypeErrorModalOpen, setFileTypeErrorModalOpen] = useState(false);
 
+    const toggleModal = () => {
+        setFileTypeErrorModalOpen(!fileTypeErrorModalOpen);
+    };
     const toggleProcessing = () => {
         setIsProcessing(!isProcessing);
     };
 
+    const handleEngineChange = (selectedValue: string) => {
+        const selectedEngine = selectedValue as AvailableEngines;
+        setConversionEngine(selectedEngine);
+        setEngineSelected(true);
+    };
+
+    const validateFileType = () => {
+        if (fileToConvert) {
+            const fileExtension =
+                fileToConvert.name.split(".").pop()?.toLowerCase() || "";
+            if (
+                validFileExtensions[conversionProcessingData.engineType] ===
+                fileExtension
+            ) {
+                setIsProcessing(!isProcessing);
+                return;
+            }
+        }
+        setFileTypeErrorModalOpen(true);
+    };
+
     // TODO: use conversion template data to render the form
     console.log("conversion form data", conversionProcessingData);
     const conversionForm = (
         <div className={classNames(styles.container, theme.lightTheme)}>
+            {fileTypeErrorModalOpen ? (
```

Comment:
```suggestion
            {fileTypeErrorModalOpen && (
```
and remove the `:null`. It's shorthand for what you have here. You could also do that for line 102

Reply:
Ok